src/content_script/input_tools.js

```diff
@@ -0,0 +1,237 @@
+      const left =
+        anchorCoords.left +
+        pageXOffset +
+        anchor.offsetWidth +
+        buttonXOffset -
+        InputTools.dimensionToInt(anchorStyle.paddingRight) +
+        "px";
+
+      const top = anchorCoords.top + pageYOffset + "px";
+
+      if (isFixed) {
+        elem.style.position = "fixed";
+      }
+
+      if (elem.style.left !== left) {
+        elem.style.left = left;
+      }
+
+      if (elem.style.top !== top) {
+        elem.style.top = top;
+      }
+    }
+
+    intervalId = setInterval(updatePosition, 200);
+    updatePosition();
+  },
+
+  async handleOnClickSLButton(inputElem, slButton) {
+    if (InputTools.isLoading) {
+      return;
+    }
+    InputTools.isLoading = true;
+    slButton.classList.add("loading");
+
+    let res = await sendMessageToBackground("NEW_RANDOM_ALIAS");
+    if (res.error) {
+      alert("SimpleLogin Error: " + res.error);
+      res = { alias: "" };
+    }
+
+    InputTools.isLoading = false;
+    slButton.classList.remove("loading");
+
+    inputElem.value = res.alias;
+  },
+
+  sumPixel(dimensions) {
+    let sum = 0;
+    for (const dim of dimensions) {
+      sum += InputTools.dimensionToInt(dim);
+    }
+
+    return sum;
+  },
+
+  dimensionToInt(dim) {
+    try {
+      const pixel = parseFloat(dim.replace(/[^0-9.]+/g, ""));
+      return isNaN(pixel) ? 0 : pixel;
+    } catch (e) {
+      return 0;
+    }
+  },
+};
+
+const MutationObserver =
+  window.MutationObserver ||
+  window.WebKitMutationObserver ||
+  window.MozMutationObserver;
+
+/**
+ * Add DOM mutations listener
+ */
+function addMutationObserver() {
+  const mutationObserver = new MutationObserver(function (mutations) {
+    mutations.forEach(function (mutation) {
+      for (const addedNode of mutation.addedNodes) {
+        // add SLButton for newly added nodes
+        InputTools.init(addedNode);
+      }
+
+      for (const removedNode of mutation.removedNodes) {
+        // destroy SLButton for removed nodes
+        InputTools.destroy(removedNode);
+      }
+    });
+  });
+
+  const target = document.body;
+  if (!target) return;
+
+  mutationObserver.observe(target, {
+    childList: true,
+    subtree: true,
+  });
+}
+
+/**
+ * Send message to background.js and resolve with the response
+ * @param {string} tag
+ * @param {object} data
+ */
+function sendMessageToBackground(tag, data = null) {
+  return new Promise((resolve) => {
+    try {
+      chrome.runtime.sendMessage(
+        {
+          tag,
+          data,
+        },
+        function (response) {
+          resolve(response);
+        }
+      );
+    } catch (e) {
+      // Extension context invalidated.
+      alert("SimpleLogin Error: Please reload the page.");
+    }
+  });
+}
+
+async function initModule() {
+  SLSettings = await sendMessageToBackground("GET_APP_SETTINGS");
+  if (SLSettings.showSLButton) {
+    InputTools.init(document);
+    addMutationObserver();
+  }
+}
+
+initModule();
```

Comment:
should we add `await` here?

Reply:
No, because `await` can't be used in global context